app/Exceptions/Auth/TokenGenerationException.php

```diff
@@ -0,0 +1,24 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Exceptions\Auth;
+
+use Exception;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Request;
+use Illuminate\Http\Response;
+
+final class TokenGenerationException extends Exception
+{
+    public function render(Request $request): JsonResponse
+    {
+        return new JsonResponse(
+            [
+                'error' => 'There was an error registering the user with the provided details',
```

Comment:
Is this the correct message, as it seems to be the same message as the RegistrationException below?

Reply:
So ive gone with a more generic error message from a safety point of view. but I think you're right and ill change it to a message that marries up to the exception itself